File: packages/dotnet/Db/Models/Product.cs
namespace LuckyDay.Db.Models;

public class Product
{
  public string Id { get; set; } = Guid.NewGuid().ToString();
  public string Name { get; set; } = null!;
  public string Description { get; set; } = null!;
  public string Price { get; set; } = null!;
  public Gender Gender { get; set; }
  public ProductType ProductType { get; set; }
  public bool IsActive { get; set; } = false;
  public DateTime DateCreated { get; set; } = DateTime.UtcNow;
  public DateTime? DateUpdated { get; set; }

  // Navigation properties
  public ICollection<ProductInventory> ProductInventories { get; set; } = [];
  public ICollection<ProductSale> ProductSales { get; set; } = [];
  public ICollection<ProductColor> ProductColors { get; set; } = [];
}

File: apps/web/src/lib/models/product.ts
import type { Gender } from './gender';
import type { ProductType } from './product-type';

export type ProductModel = {
  id: string;
  name: string;
  description: string;
  price: string;
  gender: Gender;
  productType: ProductType;
  isActive: boolean;
  dateCreated: string;
  dateUpdated: string | null;
};

export type ProductCreateModel = Pick<
  ProductModel,
  'name' | 'description' | 'price' | 'gender' | 'productType' | 'isActive'
>;

export type ProductUpdateModel = Pick<
  ProductModel,
  'name' | 'description' | 'price' | 'gender' | 'productType' | 'isActive'
>;

export type ProductResponseModel = Pick<
  ProductModel,
  | 'id'
  | 'name'
  | 'description'
  | 'price'
  | 'gender'
  | 'productType'
  | 'isActive'
  | 'dateCreated'
  | 'dateUpdated'
>;
